pkg/guardian/redis_conf_store.go

```diff
@@ -737,141 +729,78 @@ func (rs *RedisConfStore) pipelinedFetchConf() fetchConf {
 		rs.logger.WithError(err).Warnf("error send HKEYS for key %v", redisIPWhitelistKey)
 	}
 
-	if limitCount, err := limitCountCmd.Uint64(); err == nil {
-		newConf.limitCount = &limitCount
-	} else {
-		rs.logger.WithError(err).Warnf("error sending GET for key %v", redisLimitCountKey)
-	}
-
-	if limitDurationStr, err := limitDurationCmd.Result(); err == nil {
-		limitDuration, err := time.ParseDuration(limitDurationStr)
-		if err != nil {
-			rs.logger.WithError(err).Warnf("error parsing limit duration")
-		} else {
-			newConf.limitDuration = &limitDuration
-		}
-	} else {
-		rs.logger.WithError(err).Errorf("error sending GET for key %v", redisLimitDurationKey)
-	}
-
-	if limitEnabledStr, err := limitEnabledCmd.Result(); err == nil {
-		limitEnabled, err := strconv.ParseBool(limitEnabledStr)
-		if err != nil {
-			rs.logger.WithError(err).Warnf("error parsing limit enabled")
-		} else {
-			newConf.limitEnabled = &limitEnabled
-		}
-	} else {
-		rs.logger.WithError(err).Errorf("error sending GET for key %v", redisLimitEnabledKey)
-	}
-
-	if reportOnlyStr, err := reportOnlyCmd.Result(); err == nil {
-		reportOnly, err := strconv.ParseBool(reportOnlyStr)
-		if err != nil {
-			rs.logger.WithError(err).Warnf("error parsing report only")
-		} else {
-			newConf.reportOnly = &reportOnly
-		}
-	} else {
-		rs.logger.WithError(err).Warnf("error sending GET for key %v", redisReportOnlyKey)
-	}
-
-	if routeRateLimitsCounts, err := routeRateLimitsCountCmd.Result(); err == nil {
-		for route, countStr := range routeRateLimitsCounts {
-			parsedURL, urlParseErr := url.Parse(route)
-			count, intParseErr := strconv.ParseUint(countStr, 10, 64)
-			if urlParseErr != nil || intParseErr != nil {
-				rs.logger.WithError(urlParseErr).WithError(intParseErr).Warnf("error parsing route limit duration for %v", route)
+	if globalRateLimitBytes, err := globalRateLimitCmd.Bytes(); err == nil {
+		config := GlobalRateLimitConfig{}
+		if err := json.Unmarshal(globalRateLimitBytes, &config); err == nil {
+			if config.Version == GlobalRateLimitConfigVersion {
+				newConf.globalRateLimit = &config
```

Comment:
I think this function was structured this way because even if certain keys are missing, the other keys should still update their corresponding fields, meaning no early returns. It could be structured as:

```go
if err != nil {
    // Log error
} else {
    // Use result
}
```

if that seems clearer.

Reply:
I tend to prefer this pattern if the error is not fatal:

```go
var f Frobinator
// or
f := Frobinator(Field1: "Default value"} // set up defaults to use if getFrobinator() fails
f, err = getFrobinator()
if err != nil {
  // log error but continue
}
// use f
```